internal/pkg/template/diff/override.go

```diff
@@ -40,58 +43,185 @@ func (m *ignorer) parse(_, _ *yaml.Node, _ string, _ overrider) (diffNode, error
 		return false
 	}
-	return eqIntrinsicFunc(intrinsicFuncFullFormName(fullFormNode), shortFormNode.Tag)
+	// Exactly one of from and to is full form.
+	return (from.Kind == yaml.MappingNode || to.Kind == yaml.MappingNode) && (from.Kind != to.Kind)
 }
 
-func (_ *intrinsicFuncFullShortFormConverter) parse(from, to *yaml.Node, key string, overrider overrider) (diffNode, error) {
+// parse compares two intrinsic function nodes written in different form (full vs. short).
+// When the inputs to the intrinsic functions have different data types, parse assumes that no type conversion is needed
+// for correct comparison.
+// E.g. given "!Func: [1,2]" and "Fn::Func: '1,2'", parse assumes that comparing [1,2] with "1,2" produces the desired result.
+// Note that this does not hold for "GetAtt" function: "!GetAtt: [1,2]" and "!GetAtt: 1.2" should be considered the same.
+// parse assumes that from and to are matched by intrinsicFuncMapTagConverter.
+func (*intrinsicFuncMapTagConverter) parse(from, to *yaml.Node, key string, overrider overrider) (diffNode, error) {
 	var diff diffNode
 	var err error
 	if from.Kind == yaml.MappingNode {
 		// The full form mapping node always contain only one child node. The second element in `Content` is the 
 		// value of the child node. Read https://www.efekarakus.com/2020/05/30/deep-dive-go-yaml-cfn.html.
-		diff, err = parse(from.Content[1], stripTag(to), intrinsicFuncFullFormName(from), overrider)
+		diff, err = parse(from.Content[1], stripTag(to), intrinsicFunctions[intrinsicFuncName(from)].fullName, overrider)
 	} else {
-		diff, err = parse(stripTag(from), to.Content[1], intrinsicFuncFullFormName(to), overrider)
+		diff, err = parse(stripTag(from), to.Content[1], intrinsicFunctions[intrinsicFuncName(to)].fullName, overrider)
+	}
+	if diff == nil {
+		return nil, err
+	}
+	return &keyNode{
+		keyValue:   key,
+		childNodes: []diffNode{diff},
+	}, nil
+}
+
+// getAttConverter matches and parses two YAML nodes that calls the intrinsic function "GetAtt".
+// Unlike intrinsicFuncMapTagConverter, getAttConverter does not require "from" and "to" to be written in different form.
+// The input to "GetAtt" could be either a sequence or a scalar. All the followings are valid and should be considered equal.
+// Fn::GetAtt: LogicalID.Att.SubAtt, Fn::GetAtt: [LogicalID, Att.SubAtt], !GetAtt LogicalID.Att.SubAtt, !GetAtt [LogicalID, Att.SubAtt].
+type getAttConverter struct {
+	intrinsicFuncMapTagConverter
+}
+
+// match returns true if both from node and to node are calling the "GetAtt" intrinsic function.
+// "GetAtt" only accepts either sequence or scalar, therefore match returns false if either of from and to has invalid 
+// input node to "GetAtt".
+// Example1: "!GetAtt" and "!GetAtt" returns true.
+// Example2: "!GetAtt" and "Fn::GetAtt" returns true.
+// Example3: "!Ref" and "!GetAtt" returns false.
+// Example4: "!GetAtt [a,b]" and "Fn::GetAtt:a:b" returns false because the input type is wrong.
```

Comment:
What would happen if we don't deal with this separately 🤔 To me at least for "GetAtt" part, "!GetAtt [a,b]" and "Fn::GetAtt:a:b" seem to be identical and we only need to show them how their parameters are different from each other.

Reply:
`Fn::GetAtt: [a,b]` and `Fn::GetAtt: a.b` is identical and valid.

But  ⬇️ is invalid. `Fn::GetAtt` the intrinsic function does not accept a map like this.

```yaml
Fn::GetAtt:
   a: b
```

Therefore, `match` shouldn't return true in this case, because either the user wrote something invalid, or `Fn::GetAtt` is not meant to be an intrinsic function, and is instead a key name that happens to have the same name as the intrinsic func.

Reply:
I think the gist here is that I try to make the radius of these overriders as small as possible. 

I think of this in terms of false positive vs. false negative. A false positive is "when there is no diff but we say there is", and false negative is "when there is a diff but we fail to detect it".

From a user's perspective, it's annoying to see false positive, but it's dangerous if there are a lot of false negative (because they may have missed some important diff)

These overriders are written to reduce false positive, but they must not increase false negative.  This is why i try to make `match` as strict as possible.